node/__tests__/utils/jsonToQuerystring.test.ts

```diff
@@ -0,0 +1,46 @@
+import jsonToQuerystring from '../../utils/jsonToQuerystring'
+
+describe('json to querystring', () => {
+  it('should return null for null objects', () => {
+    expect(jsonToQuerystring('request', null)).toBeNull()
+  })
+
+  it('should convert an array into a querystring', () => {
+    const testArray = ['foo', 'bar']
+    const key = 'myKey'
+
+    expect(jsonToQuerystring(key, testArray)).toBe('myKey[0]=foo&myKey[1]=bar')
+  })
+
+  it('should convert an object into a querystring', () => {
+    const testObj = { foo: 'foo', bar: 'bar' }
+    const key = 'myKey'
+
+    expect(jsonToQuerystring(key, testObj)).toBe('myKey.foo=foo&myKey.bar=bar')
+  })
+
+  it('should convert an object with nested objects, arrays, null fields and undefined fields into a querystring', () => {
+    const testObj = {
+      items: [
+        { id: '1', quantity: 1, seller: '1' },
+        { id: '2', quantity: 1, seller: '2' },
+      ],
+      shippingData: {
+        logisticsInfo: [{ regionId: 'v2.3C4B555127ED49C492010704E5A0417F' }],
+      },
+      marketingData: {
+        utmSource: 'UTMSOURCE',
+        utmCampaign: 'UTMCAMPAIGN',
+      },
+      isCheckedIn: false,
+      undefinedParameter: undefined,
+      nullParameter: null,
```

Comment:
I'd also consider creating a test for a **nested** null parameter to be safe about requests without regionId, which are frequent.

So, this:
```
{
  ...
  shippingData: {
      logisticsInfo: [{ regionId: null }],
  }
}
```

Should generate the parameter `&request.shippingData.logisticsInfo[0].regionId=` or completely omit the property (including the `shippingData` part of it).

The first option is recommended if you want to keep the same semantics as the POST requests we do nowadays – if there's no specific value, it always sends the "regionId" as null.

I'm highlighting this because the `nullParameter: null` was completely omitted, which is not a problem, but won't be the same and I don't know how it would be while nested.